define/types.go

```diff
@@ -100,6 +100,8 @@ type Secret struct {
 
 // BuildOutputOptions contains the the outcome of parsing the value of a build --output flag
 type BuildOutputOption struct {
+	Type     string
+	Attrs    map[string]string
```

Comment:
Why is this adding an untyped map instead simple Go fields that exactly represent what we need?

Reply:
This was referenced from buildx but the idea remains the same, for every type, there are (and would be in the future) multiple arguments/options which are consolidated under attributes here. It's possible for us to have individual fields here but those would eventually proliferate into a heterogenous struct type and I'm not sure how ideal that would be.

Reply:
This is ultimately up to Buildah maintainers, not me,

---

IMHO individual fields, or individual types, is much better than `Attrs`, because `Attrs` is _equally_ heterogenous, just with much worse compiler / linter support.

We can do interfaces, type checks, wrappers, …